### PR TITLE
Ilb 23 UI api integration for vendor listing

### PR DESCRIPTION
**Pull request by NevinJoseph22**

### CODE REVIEW

File: src/app/app.config.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { provideRouter } from '@angular/router';
 import { routes } from './app.routes';
 import { provideClientHydration } from '@angular/platform-browser';
 import { provideAnimations } from '@angular/platform-browser/animations';
+// import { provideHttpClient } from '@angular/common/http';
```
Comment (kavyamolr) [inline src/app/app.config.ts]: remove commented code

File: src/app/features/vendor/services/vendor.service.ts [hunk 2]
```diff
@@ -1,57 +1,69 @@
...
+} from '../models/vendor.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class VendorService {
-
-  private baseUrl = 'http://localhost:5255/api';  
+  
+    private baseUrl = 'http://localhost:5255/api';
```
Comment (kavyamolr) [inline src/app/features/vendor/services/vendor.service.ts]: remove unwanted space. check every file

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts [hunk 3]
```diff
@@ -31,60 +31,21 @@ import { IVendor } from '../models/vendor.model';
...
 })
 export class VendorListingComponent implements OnInit {
-  vendors!: any[];
-
- constructor(private router : Router){}
  
+  vendors!: IVendor[];
+
+  constructor(private vendorService: VendorService) {}
   ngOnInit() {
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.ts:39]: empty line above ngoninit

File: src/app/app.config.ts (hunk 1 above)
Comment (NevinJoseph22) [reply src/app/app.config.ts]: done

File: src/app/features/vendor/services/vendor.service.ts (hunk 2 above)
Comment (NevinJoseph22) [reply src/app/features/vendor/services/vendor.service.ts]: done

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts (hunk 3 above)
Comment (NevinJoseph22) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.ts:39]: done

**Review by kavyamolr — APPROVED** [review]

**Review by athul-sanjeevan — APPROVED** [review]

**Pull request merged**